lib/solidus_paypal_braintree/engine.rb

```diff
@@ -10,7 +10,7 @@ class Engine < Rails::Engine
 
     initializer "register_solidus_paypal_braintree_gateway", after: "spree.register.payment_methods" do |app|
       app.config.spree.payment_methods << SolidusPaypalBraintree::Gateway
-      Spree::PermittedAttributes.source_attributes << :nonce
+      Spree::PermittedAttributes.source_attributes.concat [:nonce, :payment_type]
```

Comment:
>I need to whitelist it, otherwise I can not pass it into the controller from the view where the value needs top be set, as we don't know the type upfront.

I meant that the value isn't whitelisted. I can insert "lol" into this field and it would save correctly. We should probably add an inclusion validation to the model to prevent this.

>We don't know the type upfront. I didn't find any code that guarantees this. Maybe you can point me to?

Once we have the token, `payment_method.braintree.payment_method.find(source.token).is_a?(Braintree::PaypalAccount)`

As we only use this for display purposes, I no longer think that's worth doing though.